Mathlib/CategoryTheory/Yoneda.lean

```diff
@@ -643,6 +665,65 @@ lemma isIso_iff_isIso_yoneda_map {X Y : C} (f : X ⟶ Y) :
   rw [isIso_iff_yoneda_map_bijective]
   exact forall_congr' fun _ ↦ (isIso_iff_bijective _).symm
 
+/-- Yoneda's lemma as a bijection `(uliftYoneda.{w}.obj X ⟶ F) ≃ F.obj (op X)`
+for any presheaf of type `F : Cᵒᵖ ⥤ Type (max w v₁)` for some
+auxiliary universe `w`. -/
+@[simps! -isSimp]
+def uliftYonedaEquiv {X : C} {F : Cᵒᵖ ⥤ Type (max w v₁)} :
+    (uliftYoneda.{w}.obj X ⟶ F) ≃ F.obj (op X) where
+  toFun τ := τ.app (op X) (ULift.up (𝟙 _))
+  invFun x :=
+    { app Y y := F.map y.down.op x
+      naturality Y₁ Y₂ f := by ext; simp [uliftYoneda] }
+  left_inv τ := by
+    ext ⟨Y⟩ ⟨y⟩
+    dsimp [uliftYoneda]
+    rw [← FunctorToTypes.naturality]
+    dsimp
+    rw [Category.comp_id]
+  right_inv x := by simp
+
+attribute [simp] uliftYonedaEquiv_symm_apply_app
+
+lemma uliftYonedaEquiv_naturality {X Y : Cᵒᵖ} {F : Cᵒᵖ ⥤ Type max w v₁}
+    (f : uliftYoneda.{w}.obj (unop X) ⟶ F)
+    (g : X ⟶ Y) : F.map g (uliftYonedaEquiv.{w} f) =
+      uliftYonedaEquiv.{w} (uliftYoneda.map g.unop ≫ f) := by
+  dsimp
+  rw [uliftYonedaEquiv_apply, uliftYonedaEquiv_apply,
+    ← FunctorToTypes.naturality _ _ f g (ULift.up (𝟙 _))]
+  simp [uliftYoneda]
+
+lemma uliftYonedaEquiv_comp {X : C} {F G : Cᵒᵖ ⥤ Type max w v₁}
+    (α : uliftYoneda.{w}.obj X ⟶ F) (β : F ⟶ G) :
+    uliftYonedaEquiv.{w} (α ≫ β) = β.app _ (uliftYonedaEquiv α) :=
+  rfl
+
+@[reassoc]
+lemma uliftYonedaEquiv_symm_map {X Y : Cᵒᵖ} (f : X ⟶ Y) {F : Cᵒᵖ ⥤ Type max w v₁}
+    (t : F.obj X) :
+    uliftYonedaEquiv.{w}.symm (F.map f t) =
+      uliftYoneda.map f.unop ≫ uliftYonedaEquiv.symm t := by
+  obtain ⟨u, rfl⟩ := uliftYonedaEquiv.surjective t
+  rw [uliftYonedaEquiv_naturality]
+  simp
+
+@[simp]
+lemma uliftYonedaEquiv_uliftYoneda_map {X Y : C} (f : X ⟶ Y) :
+    DFunLike.coe (β := fun _ ↦ ULift.{w} (X ⟶ Y))
```

Comment:
Do we really need this in the lhs of the lemma?

Reply:
Yes, because otherwise the `simpNF` linter would complain.